cmd/node/main.go

```diff
@@ -383,8 +391,30 @@ func getSuite(config *config.Config) (crypto.Suite, error) {
 
 func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 	log.Trace("startNode called")
+	workingDir := getWorkingDir(ctx, log)
+
+	var err error
+
+	withLogFile := ctx.GlobalBool(logFile.Name)
+	if withLogFile {
+		var fileForLogs *os.File
+		fileForLogs, err = createLogFile(workingDir)
+		if err != nil {
+			return fmt.Errorf("%w creating a log file", err)
+		}
+
+		defer func() {
+			_ = fileForLogs.Close()
+		}()
+
+		err = logger.AddLogObserver(fileForLogs, &logger.PlainFormatter{})
+		if err != nil {
+			return fmt.Errorf("%w adding file log observer", err)
+		}
+	}
+
```

Comment:
Might want to put this in a function called `redirectNodeOutputToLogfile()`. The function `startNode()` is already quite long.

Reply:
this is not possible as I need to create the defer function inside startNode (as to be able to be executed when startNode ends) not when the redirectNodeOutputToLogfile ends.